patroni/postgresql.py

```diff
@@ -173,14 +180,32 @@ def pg_ctl(self, cmd, *args, **kwargs):
         pg_ctl = [self._pgcommand('pg_ctl'), cmd]
         if cmd in ('start', 'stop', 'restart'):
             pg_ctl += ['-w']
-            timeout = self.config.get('pg_ctl_timeout')
+            timeout = kwargs.pop('timeout', None)
+            timeout = self.config.get('pg_ctl_timeout', timeout)
```

Comment:
`pg_ctl_timeout` works for 'start', 'stop' and 'restart' cases.
If this value is not defined in the Patroni configuration file, then the default value from `pg_ctl` is used (60 seconds).

Can we somehow try to simplify everything? I mean we can use `pg_ctl start -w -t %master_start_timeout%` and later do a quick check that it is really accepting connections with `pg_isready`.
But from another side, there might be the case when one of the backends crashed and postmaster terminates all remaining  backends and does crash recovery. Such recovery sometime can last long and we should be able to release leader lock in this case as well.

And one more thing: in some cases even `pg_ctl stop` can "fail" to stop postgres withing a given timeout...